rero_ils/modules/loans/utils.py

```diff
@@ -136,5 +137,17 @@ def can_be_requested(loan):
     """Check if record can be requested."""
     if not loan.get('item_pid'):
         raise Exception('Transaction on document is not implemented.')
+
+    # check if circulation_policy allows request
     policy = get_circ_policy(loan)
-    return policy.get('allow_requests')
+    if not policy.get('allow_requests'):
+        return False
+
+    # check if location allows request
+    location_pid = loan.location_pid
+    if location_pid:
```

Comment:
Do you still need `if location_pid`? because you already return the `item.holdings_location_pid` if does not exit.

Reply:
`location.pid` property could return `None` (It should never happens... but it's a possibility). This check catch this use case.